src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -168,6 +177,24 @@ private bool TryExtractStringConcatArgs(BoundExpression lowered, out ImmutableAr
                             return true;
                         }
 
+                        // Faced `string.Concat(ReadOnlySpan<char>, ReadOnlySpan<char>) + string` (e.g. as a result of a previous rewrite).
+                        // We can only unwrap this if we can later rewrite it to `string.Concat(ReadOnlySpan<char>, ReadOnlySpan<char>, ReadOnlySpan<char>)`
+                        if ((object)method == _compilation.GetWellKnownTypeMember(WellKnownMember.System_String__Concat_ReadOnlySpanReadOnlySpan) &&
+                            TryGetWellKnownTypeMember<MethodSymbol>(lowered.Syntax, WellKnownMember.System_String__Concat_ReadOnlySpanReadOnlySpanReadOnlySpan, out _, isOptional: true))
+                        {
+                            arguments = boundCall.Arguments;
+                            return true;
```

Comment:
>return true; [](http://example.com/codeflow?start=28&length=12)

Unit-test `Test3` from my other comment hits this code path, but we never even try to use ```WellKnownMember.System_String__Concat_ReadOnlySpanReadOnlySpanReadOnlySpan```. Therefore, checking its presence here (in the enclosing 'if' condition) feels premature.